pkg/kubectl/cmd/create_clusterrole.go

```diff
@@ -0,0 +1,286 @@
+	for _, r := range resources {
+		sections := strings.Split(r, ".")
+
+		if len(sections) == 1 {
+			c.Resources = append(c.Resources, schema.GroupVersionResource{Resource: r})
+		} else {
+			c.Resources = append(c.Resources, schema.GroupVersionResource{Resource: sections[0], Group: strings.Join(sections[1:], ".")})
+		}
+	}
+
+	// Remove duplicate resource names.
+	resourceNames := []string{}
+	for _, n := range c.ResourceNames {
+		if !arrayContains(resourceNames, n) {
+			resourceNames = append(resourceNames, n)
+		}
+	}
+	c.ResourceNames = resourceNames
+
+	// Remove duplicate non-resource URLs.
+	nonResourceURLs := []string{}
+	for _, n := range c.NonResourceURLs {
+		if !arrayContains(nonResourceURLs, n) {
+			nonResourceURLs = append(nonResourceURLs, n)
+		}
+	}
+	c.NonResourceURLs = nonResourceURLs
+
+	return nil
+}
+
+func (c *CreateClusterRoleOptions) Validate(f cmdutil.Factory) error {
+	if c.Name == "" {
+		return fmt.Errorf("name must be specified")
+	}
+
+	// validate verbs.
+	containsResourceVerbs, containsNonResourceVerbs := false, false
+	for _, v := range c.Verbs {
+		if v == "*" {
+			containsResourceVerbs = true
+			containsNonResourceVerbs = true
+			break
+		}
+
+		validVerb := false
+
+		// Since there are overlapping verbs between resource verbs and non resource verbs, need to verify separately.
+		if arrayContains(validResourceVerbs, v) {
+			containsResourceVerbs = true
+			validVerb = true
+		}
+
+		if arrayContains(validNonResourceVerbs, v) {
+			containsNonResourceVerbs = true
+			validVerb = true
+		}
+
+		if !validVerb {
+			return fmt.Errorf("invalid verb: '%s'", v)
+		}
+	}
+
+	if !containsNonResourceVerbs && len(c.NonResourceURLs) > 0 {
+		return fmt.Errorf("at least one non-resource verb must be specified when non-resource URL(s) provided")
+	}
+
+	if !containsResourceVerbs && len(c.Resources) > 0 {
+		return fmt.Errorf("at least one resource verb must be specified when resource(s) provided")
+	}
+
+	// validate resources.
+	mapper, _ := f.Object()
+	for _, r := range c.Resources {
+		_, err := mapper.ResourceFor(r)
+		if err != nil {
+			return err
+		}
+	}
+
+	// validate resource names, can not apply resource names to multiple resources.
+	if len(c.ResourceNames) > 0 && len(c.Resources) > 1 {
+		return fmt.Errorf("resource name(s) can not be applied to multiple resources")
+	}
+
+	// validate resource names, at least one resource must be specified if resource name(s) provided.
+	if len(c.ResourceNames) > 0 && len(c.Resources) == 0 {
+		return fmt.Errorf("at least one resource must be specified when resource name(s) provided")
+	}
+
+	if len(c.Resources) == 0 && len(c.NonResourceURLs) == 0 {
+		return fmt.Errorf("at least one resource or non-resource URL must be specified")
+	}
+
+	return nil
+}
+
+func (c *CreateClusterRoleOptions) RunCreateClusterRole(f cmdutil.Factory, cmdOut io.Writer, cmd *cobra.Command, args []string) error {
+	mapper, _ := f.Object()
```

Comment:
@AdoHe, any code patterns need to follow here? extract the arguments from cmd object and pass them in as parameters? wonder any examples in current commands